### ISSUE TITLE
Poetry does ignores extras_require packages defined in setup.py

### ISSUE THREAD

**Issue by mr-bjerre** [python-poetry/poetry#3180]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: both debian and macOS
- **Poetry version**: 1.1.2

## Issue

If I run the following `poetry add 'camelot-py[cv]'` then I correctly get the following in `pyproject.toml`

```toml
[tool.poetry.dependencies]
python = "^3.8"
camelot-py = {extras = ["cv"], version = "^0.8.2"}
```

But `opencv-python` is not installed as stated in `extras_require` in [setup.py](https://github.com/camelot-dev/camelot/blob/master/setup.py). If I use `pip install 'camelot-py[cv]'` then it works as expected

**Comment by tamuhey** [comment]
I've confirmed this bug with `poetry add modin[dask]`

**Comment by tamuhey** [comment]
It doesn't happen with `poetry=1.0.10`

**Comment by abn** [comment]
Can you please try the fix at #3147.

### Using pipx
```sh
pipx install --suffix=@3147 'poetry @ git+https://github.com/python-poetry/poetry.git@refs/pull/3147/head'
```

### Using a container (podman | docker)
```sh
podman run --rm -i --entrypoint bash python:3.8 <<EOF
set -e
python -m pip install -q git+https://github.com/python-poetry/poetry.git@refs/pull/3147/head
python -m poetry new foobar
pushd foobar
sed -i /pytest/d pyproject.toml
python -m poetry add 'camelot-py[cv]'
EOF
```

**Comment by abn** [comment]
Duplicate of #3129
